File: library/src/main/java/ir/vasl/library/enums/ErrorCode.java
package ir.vasl.library.enums;


/**
 * Created by alishatergholi on 2/17/18.
 */
public enum ErrorCode {

    AuthorizationException(401,"Authorization Exception"),
    SessionExpire(403,"Session Expire Exception"),
    NotFound(404,"Not Found Exception"),
    InternetConnectionError(407,"Timeout Exception Internet connection Problem"),
    ServerConnectionError(500,"Server Connection Error"),
    NullPointerException(1005,"NUll Pointer Exception"),
    IOException(1006,"IO Exception"),
    RuntimeException(1007,"Runtime Exception error"),
    PermissionDenied(1009,"Permission Need for this Operation"),
    ParseException(1020,"Exception while Parse Data"),
    UnsupportedEncodingException(1020,"UnsupportedEncodingException"),
    Exception(9999,"Exception");

    private final int value;
    private String description;

    private ErrorCode(int value) {
        this.value = value;
    }

    private ErrorCode(int value, String description) {
        this.value       = value;
        this.description = description;
    }

    public int getCode() {
        return this.value;
    }

    public String getDescription() {
        return this.description;
    }

    public static ErrorCode Parse(int value) {
        if(value == 1) {
            return Exception;
        } else {
            ErrorCode[] arr$ = values();
            ErrorCode[] var2 = arr$;
            int var3 = arr$.length;

            for(int var4 = 0; var4 < var3; ++var4) {
                ErrorCode val = var2[var4];
                if(val.value == value) {
                    return val;
                }
            }
            return Exception;
        }
    }
}

File: library/src/main/java/com/github/library/enums/ErrorCode.kt
package com.github.library.enums


/**
 * Created by alishatergholi on 2/17/18.
 */
enum class ErrorCode {

    AuthorizationException(401          , "Authorization Exception"),
    SessionExpire(403                   , "Session Expire Exception"),
    NotFound(404                        , "Not Found Exception"),
    InternetConnectionError(407         , "Timeout Exception Internet connection Problem"),
    ServerConnectionError(500           , "Server Connection Error"),
    NullPointerException(601            , "NUll Pointer Exception"),
    IOException(602                     , "IO Exception"),
    RuntimeException(603                , "Runtime Exception error"),
    PermissionDenied(604                , "Permission Need for this Operation"),
    ParseDataException(605              , "Exception while parse Data"),
    UnsupportedEncodingException(606    , "UnsupportedEncodingException"),
    Exception(608                       , "Exception");

    val code: Int
    var description: String = ""

    private constructor(value: Int) {
        this.code = value
    }

    private constructor(value: Int, description: String) {
        this.code = value
        this.description = description
    }

    companion object {

        fun parse(value: Int): ErrorCode {
            if (value == 1) {
                return Exception
            } else {
                val `arr$` = values()
                val var3 = `arr$`.size

                for (var4 in 0 until var3) {
                    val `val` = `arr$`[var4]
                    if (`val`.code == value) {
                        return `val`
                    }
                }
                return Exception
            }
        }
    }
}
